h/static/scripts/group-forms/components/forms/Checkbox.tsx

```diff
@@ -0,0 +1,48 @@
+import type { CheckboxProps as BaseCheckboxProps } from '@hypothesis/frontend-shared';
+import { Checkbox as BaseCheckbox } from '@hypothesis/frontend-shared';
+import type { ComponentChildren } from 'preact';
+import { useId } from 'preact/hooks';
+
+export type CheckboxProps = Omit<BaseCheckboxProps, 'aria-describedby'> & {
+  /**
+   * Adds a description right under the checkbox and main content, aligned with
+   * the main content's left side.
+   */
+  description?: ComponentChildren;
+
+  'aria-describedby'?: string;
+};
+
+export default function Checkbox({
+  description,
+  children,
+  'aria-describedby': ariaDescribedBy,
+  ...checkboxProps
+}: CheckboxProps) {
+  const descriptionId = useId();
+
+  const describedBy: string[] = [];
+  if (ariaDescribedBy) {
+    describedBy.unshift(ariaDescribedBy);
+  }
+  if (description) {
+    describedBy.push(descriptionId);
+  }
+
+  return (
+    <div>
+      <BaseCheckbox {...checkboxProps} aria-describedby={describedBy.join(' ')}>
+        <span className="text-grey-7">{children}</span>
```

Comment:
Is this explicit color needed? It looks like it gets inherited from the parent if not specified.

Reply:
I set it to avoid the same issue we faced with the radio buttons where the main content ended up in a lighter color than the description because we were explicitly setting a color only to the description.

This ensures a slightly darker shade of grey in the main content, and follows the same pattern used in https://github.com/hypothesis/frontend-shared/pull/1955

It's a bit of a pain that we have to remember about this everywhere though. We probably need to consolidate the base font color in all apps soon-ish, together with the base font size, as you suggested in https://hypothes-is.slack.com/archives/C1M8NH76X/p1746003644996359

Reply:
Is the inner span needed here or can the class attribute go on the `BaseCheckbox` itself?